File: test/VbNet.LanguageServer.Tests/Services/SignatureHelpServiceTests.cs
using Microsoft.Extensions.Logging.Abstractions;
using VbNet.LanguageServer.Protocol;
using VbNet.LanguageServer.Services;
using VbNet.LanguageServer.Workspace;
using Xunit;

namespace VbNet.LanguageServer.Tests.Services;

/// <summary>
/// Unit tests for SignatureHelpService.
/// </summary>
public class SignatureHelpServiceTests
{
    private readonly WorkspaceManager _workspaceManager;
    private readonly DocumentManager _documentManager;
    private readonly SignatureHelpService _signatureHelpService;

    public SignatureHelpServiceTests()
    {
        _workspaceManager = new WorkspaceManager(NullLogger<WorkspaceManager>.Instance);
        _documentManager = new DocumentManager(_workspaceManager, NullLogger<DocumentManager>.Instance);
        _signatureHelpService = new SignatureHelpService(
            _workspaceManager,
            _documentManager,
            NullLogger<SignatureHelpService>.Instance);

        _workspaceManager.Initialize();
    }

    [Fact]
    public async Task GetSignatureHelpAsync_NoDocument_ReturnsNull()
    {
        var @params = new SignatureHelpParams
        {
            TextDocument = new TextDocumentIdentifier { Uri = "file:///nonexistent.vb" },
            Position = new Position { Line = 0, Character = 0 }
        };

        var result = await _signatureHelpService.GetSignatureHelpAsync(@params, CancellationToken.None);

        Assert.Null(result);
    }

    [Fact]
    public async Task GetSignatureHelpAsync_NullParams_ReturnsNull()
    {
        var result = await _signatureHelpService.GetSignatureHelpAsync(null!, CancellationToken.None);

        Assert.Null(result);
    }
}

File: test/VbNet.LanguageServer.Tests.Vb/Services/SignatureHelpServiceTests.vb
Imports System.Threading
Imports System.Threading.Tasks
Imports Microsoft.Extensions.Logging.Abstractions
Imports VbNet.LanguageServer.Protocol
Imports VbNet.LanguageServer.Services
Imports VbNet.LanguageServer.Workspace
Imports Xunit

Namespace VbNet.LanguageServer.Tests.Services

    ''' <summary>
    ''' Unit tests for SignatureHelpService.
    ''' </summary>
    Public Class SignatureHelpServiceTests
        Private ReadOnly _workspaceManager As WorkspaceManager
        Private ReadOnly _documentManager As DocumentManager
        Private ReadOnly _signatureHelpService As SignatureHelpService

        Public Sub New()
            _workspaceManager = New WorkspaceManager(NullLogger(Of WorkspaceManager).Instance)
            _documentManager = New DocumentManager(_workspaceManager, NullLogger(Of DocumentManager).Instance)
            _signatureHelpService = New SignatureHelpService(
                _workspaceManager,
                _documentManager,
                NullLogger(Of SignatureHelpService).Instance)

            _workspaceManager.Initialize()
        End Sub

        <Fact>
        Public Async Function GetSignatureHelpAsync_NoDocument_ReturnsNull() As Task
            Dim request = New SignatureHelpParams With {
                .TextDocument = New TextDocumentIdentifier With {.Uri = "file:///nonexistent.vb"},
                .Position = New Position With {.Line = 0, .Character = 0}
            }

            Dim result = Await _signatureHelpService.GetSignatureHelpAsync(request, CancellationToken.None)

            Assert.Null(result)
        End Function

        <Fact>
        Public Async Function GetSignatureHelpAsync_NullParams_ReturnsNull() As Task
            Dim result = Await _signatureHelpService.GetSignatureHelpAsync(Nothing, CancellationToken.None)

            Assert.Null(result)
        End Function
    End Class

End Namespace
